src/openrc/rc.c

```diff
@@ -413,17 +412,13 @@ handle_signal(int sig)
 		break;
 
 	case SIGINT:
-		if (!signame)
-			xasprintf(&signame, "SIGINT");
-		/* FALLTHROUGH */
+		eerrorx("%s: caught SIGINT, aborting", applet);
+		/* NOTREACHED */
 	case SIGTERM:
-		if (!signame)
-			xasprintf(&signame, "SIGTERM");
-		/* FALLTHROUGH */
+		eerrorx("%s: caught SIGTERM, aborting", applet);
```

Comment:
you can keep the previous logic without duplicating the errorx call

```
const char *signame = NULL;
...
if (!signame)
  signame = "SIGINT";
...
```

Reply:
> you can keep the previous logic without duplicating the errorx call

Is that worth it though? It's more code than just calling errorx directly (and simpler to follow since it doesn't rely on fallthroughs).

But I can update the PR if that's your preference.

Reply:
your version requires duplicating error strings & function calls.  the current version involves setting a pointer with some cascading NULL checks.  imo that is better.

Reply:
Updated in 83bba03